src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -100,39 +99,28 @@ class ToolCustomLockfile(Lockfile, _ToolLockfileMixin):
 @dataclass(unsafe_hash=True)
 class PexRequirements:
     req_strings: FrozenOrderedSet[str]
-    apply_constraints: bool
```

Comment:
> I could really use eyes on the revert of #12675 in 3e1f3bc. That was not a clean revert. I don't want to stomp the subsequent lock file work that tangled the clean revert.

I think it was better modeling to define `apply_constraints` on `PexRequirements` than `PexRequest` because it's better namespaced. But either approach is valid - not blocking. 

Not a big deal either way because I believe this current plan* is to deprecate `[python-setup].requirement_constraints` in favor of the new lockfile feature, meaning this code will be deleted.

*I know you've mentioned constraints files could still be useful to pin certain transitive deps, which I agree. But we co-opted them too much with `[python-setup].resolve_all_constraints` (on by default) that to get people to migrate to lockfiles, deprecating constraints files seems the clearest path. And then maybe we retcon the mechanism after it was removed to behave how you intend. (This plan is open to change.)

Reply:
The other reason it was moved here was that the two LockFile cases never use the constraints / are "for" the constraints, and so this field was only relevant to directly specified constraints (afaict). But I should have done it as prework, and as Eric says: not a big deal.

Reply:
Yeah, I'd prefer to keep a revert a revert and follow up with the re-improvement post cherry pick if that makes sense.

Reply:
Noting I still intend to do this but my stack got deeper here temporarily with the bug this upgrade exposed.